### PR TITLE
test: adding test for crossing hello trys with historical proofs

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

closes: #3831 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/04-channel/keeper/upgrade_test.go [hunk 1]
```diff
@@ -2150,3 +2150,79 @@ func (suite *KeeperTestSuite) TestSyncUpgradeSequence() {
...
+			err = path.EndpointB.ChanUpgradeInit()
+			suite.Require().NoError(err)
+
+			suite.coordinator.CommitBlock(suite.chainA, suite.chainB)
+
+			err = path.EndpointB.UpdateClient()
+			suite.Require().NoError(err)
+
+			// TODO: QueryChannelUpgradeProof queries the counterparty (EndpointA), this feels unintuitive
+			historicalChannelProof, historicalUpgradeProof, proofHeight := path.EndpointB.QueryChannelUpgradeProof()
```
Comment (damiannolan) [inline modules/core/04-channel/keeper/upgrade_test.go:2200]: Right now, `path.EndpointA.QueryChannelUpgradeProof()` is querying the counterparty of endpointB (i.e. chain A). 

I feel like this is slightly unintuitive as it reads as if you are querying endpointB for the proof but in reality whats happening is we are querying for proofs which we would like to verify on endpointB(chainB).

I'd like to adapt this in a follow up PR.

Comment (damiannolan) [reply modules/core/04-channel/keeper/upgrade_test.go:2200]: Opened #5147

**Review by chatton — APPROVED** [review]
Really nice work on this one, this is a really tricky situation to set up and the test is super readable 🥇

File: modules/core/04-channel/keeper/upgrade_test.go (hunk 1 above)
Comment (damiannolan) [reply modules/core/04-channel/keeper/upgrade_test.go]: addressed [1c975b9](https://github.com/cosmos/ibc-go/pull/5143/commits/1c975b9ba0dfdb4293a2c02f9e1a0822bf4d063c)

**Pull request merged**